lib/Parser/BisonTokens.cpp

```diff
@@ -0,0 +1,29 @@
+// A macro to convert a symbol to a string
+#define STRINGIFY(x) #x
+
+// Redefine TOKEN to accept either a single argument (and deduce the literal via
+// tokenization) or two arguments (the second being the desired literal).
+#define _TOKEN_IMPL(NAME, STR) %token <glu::Token> NAME __COUNTER__ STR
+#define _GET_TOKEN_MACRO(_1, _2, NAME, ...) NAME
+#define TOKEN1(NAME) _TOKEN_IMPL(NAME, STRINGIFY(NAME))
+#define TOKEN2(NAME, STR) _TOKEN_IMPL(NAME, STR)
+#define TOKEN(...) _GET_TOKEN_MACRO(__VA_ARGS__, TOKEN2, TOKEN1)(__VA_ARGS__)
```

Comment:
there was a simpler way...
```suggestion
#define TOKEN(NAME) _TOKEN_IMPL(NAME, STRINGIFY(NAME))
#define TOKEN2(NAME, STR) _TOKEN_IMPL(NAME, STR)
```
and just use TOKEN2() each time you have two arguments, for all the defines under, instead of having a complicated overload